### ISSUE TITLE
Utilization rate of the gpu

### ISSUE THREAD

**Issue by shatealaboxiaowang** [mravanelli/SincNet#6]
my data is very very big, Utilization rate of the gpu(24G) is very low when trainning, 128 batch size used 1G of 24G, I try to add batch size to improve the utilization rate, But I worry about the impact on convergence, do you have some way which can improve the utilization rate of gpu? thank you

**Comment by mravanelli** [comment]
Dear Wenkai,
increasing the batch size and the size of your DNN are good ways to better
use your GPU memory. In general, using low batch sizes is quite helpful for
the system performance, due to a regularization effect that arises from
"noisy" gradients. In your case you can try to do the following:
- increase the batch_size
- increase the network size
- increase the dropout rate (the have more regularization)
This isn't a gold recipe, but at least is something one can try!

**Comment by shatealaboxiaowang** [comment]
Thank you for your reply, you are so friendly, your sincNet is so great, and i have anothor question, my sincNet present parameters configuration is:  my data set has about 4000 speakers(as 4000 classes) and each speaker has a 20-second wav audio clip ;length of sinc filter is 125(yours is 251) and input size of sincNet is 1600(yours if 3200) because of my voice sample rate is 8000(yours is 16000); the d-vector size(the number of neurons in the penultimate layer) is 400(yours is 2048); my question is that can i change the number of filters in CNN_net(yours is 80, 60, 60) and the filters' length(yours is 5) of the second and third layer in CNN_net? do you have some gold recipes for my configuration? thank you very much!

**Comment by mravanelli** [comment]
Hi Wenkai,
unfortunately I don't have any gold recipe. The best you can do is to play
a bit with all the hyperparamters and find a configuration that works well
for you (this is computational-demanding, but it is the standard approach
with current deep learning techniques). In my case, the hyperparameters
that you find in the config file were pretty good for the task I addressed
in my paper.

**Comment by shatealaboxiaowang** [comment]
Still thank you for your enthusiasm and your open source code!

**Comment by mravanelli** [comment]
You are welcome! Let me know if you have other questions.

**Comment by shatealaboxiaowang** [comment]
I  have another question，have you tried to use thousands of classes（speakers）to training SincNet？I have tried 4000 classification，but the test_err is so high （about 0.5）。the number of classes is about 420 in your training set and test_err is very low。do you have some gold recipes for thousands of classes to training？If too many classes will cause adverse effect to feature extraction（d -vector）？thank you！

**Comment by shatealaboxiaowang** [comment]
I  have another question，have you tried to use thousands of classes（speakers）to training SincNet？I have tried 4000 classification，but the test_err is so high （about 0.5）。the number of classes is about 420 in your training set and test_err is very low。do you have some gold recipes for thousands of classes to training？If too many classes will cause adverse effect to feature extraction（d -vector）？thank you！
